### DISCUSSION TITLE
How to paginate expanded query

### DISCUSSION THREAD

**Discussion by zepdrix** [swellstores/community#362] [Help] [answered]
I'm trying to use `swell-node` to find all variants of a product by searching the parent product's sku. I'm expanding the query with variants so the request likes like this:
```
swell.get('/products', {
  where: { sku: productId },
  expand: 'variants:50',
})
```

The problem is, if there are more variant results than the limit per page, I don't know how to paginate through this expanded field.

```
    [
      {
        name: 'book',
        sku: '1',
        ...
        id: '64f1f7261a42800012c52d07',
        variants: { count: 20, page_count: 4, page: 1, results: [Array] }
      }
    ]
```

How can I alter my original query to get page 4 of the expanded variants field in the result above?

Thanks!

**Comment by logeshswell** [comment]
@zepdrix Can you try the below query?
```ts
swell.get('/products', {
  where: { sku: 'product-sku' },
  fields: ['variants'],
  include: {
    variants: {
      url: '/products:variants',
      data: {
        active: true,
        // ...
        limit: 10,
        page: 2 // pagination
      },
      params: {
        parent_id: 'id'
      }
    }
  }
});
```

**Reply by zepdrix** [reply]
thanks @logeshswell but this did not work, the result i got from this is:
```
{
      count: 1,
      page_count: 1,
      page: 1,
      results: [ { id: '64f1f7261a42800012c52d07' } ]
    }
```
i only see a product's id here in the results, there's nothing referring to the variant

**Reply by logeshswell** [reply]
Can you try the same query in Developer console?
Also, can you help me with your store_id?

**Reply by zepdrix** [reply]
Thanks, I tried this in the developer console and it worked but I'm still not sure how to get it working with `swell-node`, my store_id is `lu-test-2`, thanks!

**Reply by logeshswell** [reply, marked as answer]
I tried running the same code in your swell store and it is working fine for me.
Tested in node(v18.16.0) and swell-node(v5.2.0)
```js
const swell = require('swell-node');
swell.init('lu-test-2', 'sk_test_51Ki****KH3y');

const resp = await swell.get('/products', {
  where: { sku: '1' },
  fields: ['variants'],
  include: {
    variants: {
      url: '/products:variants',
      data: {
        active: true,
        limit: 10,
        page: 2
      },
      params: {
        parent_id: 'id'
      }
    }
  }
});
console.log(resp);
```

**Reply by zepdrix** [reply]
Got it thank you, this does work, I see all the variants. Is there any way to also have the parent product information be populated? Right now only the `id` is coming through. Do I need to list them all in the `fields` array?

**Reply by logeshswell** [reply]
You can remove the line `fields: ['variants'],` to print the full information of the product.
